security/alicrypto/csky/hobbit12_evb/aes.c

```diff
@@ -0,0 +1,345 @@
+/**
+ * Copyright (C) 2017 The YunOS Project. All rights reserved.
+ */
+
+#include "mbed_crypto.h"
+#include "ali_crypto.h"
+#include "drv_tee.h"
+
+static uint8_t key_copy[32];
+static uint32_t keylen_copy;
+
+ali_crypto_result ali_aes_get_ctx_size(aes_type_t type, size_t *size)
+{
+    if (size == NULL) {
+        PRINT_RET(ALI_CRYPTO_INVALID_ARG, "aes_get_ctx_size: bad input!\n");
+    }
+    switch(type) {
+        case AES_ECB:
+        case AES_CBC:
+        case AES_CTR:
+#if defined(MBEDTLS_CIPHER_MODE_CFB)
```

Comment:
Why have mbedtls macro here?

Reply:
Current, our HW engine support AES ECB & CBC only.. we still want to run other application populated in AliOS things which might implement AES other mode such as CFB..